Index: src/components/page/cart.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/components/page/cart.js	(revision 21f84237e14da6832e0598f6a0dbdee372753049)
+++ src/components/page/cart.js	(date 1583852516417)
@@ -11,8 +11,8 @@
         stripe: null,
     }
     componentDidMount() {
-        const stripe = window.Stripe("pk_live_OGxNOUzWvpoUJS3yscyZ6Ccw00ukIopzD4")
-        /*const stripe = window.Stripe("pk_test_4xqQzlAyU2e9MJ2h9P1SapFe00K4jXy6Rk")*/
+        /*const stripe = window.Stripe("pk_live_OGxNOUzWvpoUJS3yscyZ6Ccw00ukIopzD4")*/
+        const stripe = window.Stripe("pk_test_4xqQzlAyU2e9MJ2h9P1SapFe00K4jXy6Rk")
         this.setState({ stripe })
     }
 
Index: gatsby-node.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- gatsby-node.js	(revision 21f84237e14da6832e0598f6a0dbdee372753049)
+++ gatsby-node.js	(date 1583852502717)
@@ -1,6 +1,6 @@
-/*require("dotenv").config({
+require("dotenv").config({
     path: `.env.${process.env.NODE_ENV}`,
-})*/
+})
 
 // Implement the Gatsby API “onCreatePage”. This is
 // called after every page is created.
Index: content/categories/main.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- content/categories/main.json	(revision 21f84237e14da6832e0598f6a0dbdee372753049)
+++ content/categories/main.json	(date 1583852486246)
@@ -2,7 +2,7 @@
   "name": "Necessities!",
   "id": "main",
   "products": [
-    "sku_GqieRpegmpEELC","sku_Gr3s24Ebbr2EkP"
+    "sku_Gs8qHKkVxMdXfv"
   ],
   "image":  "content/assets/blue-and-red-plants-2505693.jpg"
 }
